python.asciidoc

```diff
@@ -0,0 +1,562 @@
+        return 2 * (self._width + self._height)
+
+    def area(self):
+        return self._width * self._height
+
+if __name__ == "__main__":
+    for i in xrange(1, 5):
+        r = Rectangle(i, i + 2)
+
+        print("Width and height: {}".format(r.dimensions()))
+        print("Circumference: {}".format(r.circumference()))
+        print("Area: {}".format(r.area()))
+        print("")
+----
+
+In this example we have a `Rectangle` class that encapsualtes a width
+and height, and can compute a couple of attributes of the rectangle.
+
+Our program uses the Rectangle class to print these attributes for a
+sequence of ever larger rectangles.
+
+To run the program, execute these commands in the measurements
+directory:
+
+[source,shell]
+----
+$ python rectangle.py
+----
+
+And the Pony version:
+
+[source,ponylang]
+----
+use "collections"
+
+actor Main
+  new create(env: Env) =>
+    for i in Range[I32](1, 5) do
+        let r = Rectangle(i, i + 2)
+        env.out.print("Width and height: " + r.dimensions())
+        env.out.print("Circumference: " + r.circumference().string())
+        env.out.print("Area: " + r.area().string())
+        env.out.print("")
+    end
+
+class Rectangle
+  let _width: I32
+  let _height: I32
+
+  new create(width: I32, height: I32) =>
+    _width = width
+    _height = height
+
+  fun dimensions(): String => _width.string() + " " + _height.string()
+
+  fun circumference(): I32 => 2 * (_width + _height)
+
+  fun area(): I32 => _width * _height
+----
+
+In the Pony version, the `Rectangle` class structure is essentially the
+same as the Python version. The `let` keyword introduces fields or local
+variables that must be initialized exactly once. Visibility in Pony is
+controlled by the naming: elements with a leading underscore are package
+private, all others are public. This is almost identical to how the
+private variables are stored in our Python implementation.
+
+Because Pony is also a functional language, all statements are also
+expressions. The return value of functions is simply the last expression
+in the body. There's no need for an explicit `return` statement.
+
+For-loops in Pony are a lot like Python’s for loop, and is controlled by
+an intance of the `Iterator` interface. In this case we want a simple
+integer range, so we use a `Range` object parameterized with `I32` which
+gives us a sequential range of 32-bit integers.
+
+The `Range` class is in the collections package, and we import that with
+the `use` keyword.
+
+Like Python, Pony doesn’t have any implicit conversions: we have to
+invoke the `string()` method on each of the numeric values to get
+strings that we can concatenate to build the output. This is very
+similar to calling `__str__` or `str()` on our Python objects.
+
+To run the program, execute these commands in the measurements
+directory:
+
+[source,shell]
+----
+$ ponyc main.pony
+$ ./measurements
+----
+
+[[getting-functional]]
+Getting functional
+~~~~~~~~~~~~~~~~~~
+
```

Comment:
I have a quibble with this example. Pony has an `elseif` keyword, so the Pony equivalent of the Python code would probably use that.

```
    if a > b then
      "a is bigger than b"
    elseif a < b => then
      "b is bigger than a"
    else 
      "a and b are the same"
    end
```

I'd argue that the `match` statement shown in this example isn't really representative of the way that `match` is used in most Pony code. At the very least, I think the Pony `match` statement more closely corresponds to the pattern of [dictionary mapping](http://www.pydanny.com/why-doesnt-python-have-switch-case.html) that you see in Python programs.

I see that this is similar to the example from the Java chapter, and I have the same quibble there.